common/src/main/scala/org/neo4j/spark/service/MappingService.scala

```diff
@@ -122,10 +125,22 @@ class Neo4jWriteMappingStrategy(private val options: Neo4jOptions)
         val field = schema(i)
         val neo4jValue = Neo4jUtil.convertFromSpark(seq(i), field)
         neo4jValue match {
-          case map: MapValue => Neo4jUtil.flattenMap(map.asMap(), field.name)
-            .asScala
-            .map(t => (t._1, Values.value(t._2)))
-            .toSeq
+          case map: MapValue =>
+            val scalaMap = map.asMap().asScala.toMap
+            val dupKeys = scalaMap.flattenKeys(field.name)
+              .groupBy(identity)
+              .collect { case (x, List(_,_,_*)) => x }
+            if (dupKeys.nonEmpty) {
```

Comment:
actually, it's a pity to iterate over the map twice, could we move the duplicate check into `flattenMap`?

Reply:
I'm also thinking that we could actually aggregate values in a collection when there are duplicates (with the caveat that the order is not deterministic is the initial map does not guarantee any iteration order)
since it's a breaking change, could we maybe allow this behavior only if a specific config flag is set (which would default to `false` -> the default value can change at the next major version)
wdyt?

Reply:
Let's eliminate logging because if we have a Dataframe with a column map with collisions with a count of 5b we risk printing the warning 5b times, which is not super...